x-pack/plugins/fleet/server/services/agents/upgrade_action_runner.ts

```diff
@@ -148,24 +148,38 @@ export async function upgradeBatch(
 
 export const MINIMUM_EXECUTION_DURATION_SECONDS = 60 * 60 * 2; // 2h
 
-const getRollingUpgradeOptions = (startTime?: string, upgradeDurationSeconds?: number) => {
+export const getRollingUpgradeOptions = (startTime?: string, upgradeDurationSeconds?: number) => {
   const now = new Date().toISOString();
-  const expiration = moment(startTime ?? now)
-    .add(1, 'year')
-    .toISOString();
   // Perform a rolling upgrade
   if (upgradeDurationSeconds) {
+    const minExecutionDuration = Math.min(
+      MINIMUM_EXECUTION_DURATION_SECONDS,
+      upgradeDurationSeconds
+    );
     return {
       start_time: startTime ?? now,
       rollout_duration_seconds: upgradeDurationSeconds,
-      expiration,
+      minimum_execution_duration: minExecutionDuration,
+      // expiration will not be taken into account with Fleet Server version >=8.7, it is kept for BWC
+      // in the next major, expiration and minimum_execution_duration should be removed
```

Comment:
added back old expiration logic for BWC as per the latest [discussion](https://github.com/elastic/fleet-server/pull/2243#issuecomment-1375627579).